src/uct/ib/base/ib_device.h

```diff
@@ -151,15 +167,22 @@ typedef struct uct_ib_device {
 
 
 /**
- * RoCE version priorities
+ * RoCE version
  */
 typedef struct uct_ib_roce_version_desc {
-    uint8_t     roce_major;
-    uint8_t     roce_minor;
-    sa_family_t address_family;
+    /** RoCE version described by the UCT_IB_DEVICE_ROCE_* values */
```

Comment:
there is some mixup - roce "version" is sometimes the version only and sometimes it's version+addr_family
i suggest to rename uct_ib_roce_version_desc_t to uct_ib_roce_version_info_t and rename variables and parameters of this type as "uct_ib_roce_version_info_t *roce_info" in all places (maybe except roce_prio)